### PR TITLE
separate cut planes by neigh skin for balance shift

### PR DESCRIPTION
**Pull request by sjplimp**
**Summary**

Add logic to the balance and fix_balance shift commands to insure cut planes in any dimension
are separated by at least distance = neighbor skin.  This can help prevent particles from being lost when dynamics are run with proc sub-domains that are too narrow in a particular dimension.

**Related Issue(s)**

Fixes issue #2337

**Author(s)**

Steve

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

N/A.  Though if this constraint is invoked, then the results of a simulation will change, since particles will e assigned to different procs.

**Implementation Notes**

**Post Submission Checklist**

- [ ] The feature or features in this pull request is complete
- [ ] Licensing information is complete
- [ ] Corresponding author information is complete
- [ ] The source code follows the LAMMPS formatting guidelines
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] The added/updated documentation is integrated and tested with the documentation build system
- [ ] The feature has been verified to work with the conventional build system
- [ ] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.
- [ ] A package specific README file has been included or updated
- [ ] One or more example input decks are included

**Further Information, Files, and Links**

### CODE REVIEW

**Comment by sjplimp** [comment]
do not merge yet, pending examination of weights for the in.balance.var.dynamic test script by @akohlmey

File: doc/balance.rst [hunk 1]
```diff
@@ -0,0 +1,566 @@
+.. index:: balance
```
Comment (rbberger) [inline doc/balance.rst:1]: @sjplimp this file should be in doc/src

**Comment by rbberger** [comment]
@sjplimp looks good. the warnings are now gone:

Old:
https://ci.lammps.org/job/dev/job/master/job/ubuntu/job/run_tests/job/cmake_testing_mpi/320/testReport/junit/tests.test_examples/BalanceTestCase/test_balance_var_dynamic_parallel/attachments/BalanceTestCase_test_balance_var_dynamic_parallel_stdout.log

New:
https://ci.lammps.org/job/dev/job/pull_requests/job/ubuntu/job/run_tests/job/cmake_testing_mpi/147/testReport/tests.test_examples/BalanceTestCase/test_balance_var_dynamic_parallel/attachments/BalanceTestCase_test_balance_var_dynamic_parallel_stdout.log

Merge?

**Review by akohlmey — APPROVED** [review]
It will be easier for me to look into the timer based imbalance measurements going out of range after this is included.

**Pull request merged**